src/EnergyPlus/DataSurfaces.hh

```diff
@@ -177,6 +177,10 @@ namespace DataSurfaces {
         Num
     };
 
+    int constexpr iFWC_Floor = (int)FWC::Floor;
+    int constexpr iFWC_Wall = (int)FWC::Wall;
+    int constexpr iFWC_Ceiling = (int)FWC::Ceiling;
```

Comment:
How are people feeling about this? It looks similar to the way we used to do fake `enum` before we had real `enum`s, with the big difference being that these are **defined** using the actual `enum` values. Note, these are really only useful for enums that are also used as array indices, which is a small number of them.

Reply:
Not sure. It feels awkward at first glance, but it does lighten the code where they are used. It's not all that different from an `auto &` shortcut, except it's defined in the header instead of locally.

Reply:
My take is the rule of 3 (my rule). If there are 3 or more places where this would be used then it would save some clock cycles. Always converting the enum to int versus using a constexpr. I have no idea how many clock cycles. For shortcuts or something like this, that have no overhead to set up, maybe my rule should be 2 or more.

Reply:
Casting is not a clock cycle thing. The one tiny exception is `dynamic_cast` and even that is not a huge deal. C-style (i.e., `(int)`) and C++-style (i.e., `static_cast<int>()`) are just compile-time type conversions. C++-style is type-checked. C-style is not. This is just for readability.

Reference shortcuts are a little more complicated. They are also mostly for readability, but here there is also the added benefit of telling the compiler something about the equivalence and invariance of different addresses. And if they are const shortcuts, you are also telling the compiler that its okay to keep elements access through those shortcuts in registers (although it may be able to figure that out anyway) which enables further optimization.

But with casts, that doesn't apply. Casts are part of the compiler's type-checking/type-conversion system. And typing (except for floating point vs. integer, integer sign, and size i.e., float vs. double) is a high-level language construct. It doesn't exist in machine code. It only tells the compiler how to generate code, it doesn't become part of the generated code.